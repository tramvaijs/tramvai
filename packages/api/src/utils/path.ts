import path from 'node:path';

export const resolveAbsolutePathForFile = ({
  file,
  sourceDir,
  rootDir,
}: {
  file: string;
  sourceDir: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(file)) {
    return file;
  }
  if (path.isAbsolute(sourceDir)) {
    return path.join(sourceDir, file);
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, sourceDir, file);
  }
  return path.resolve(rootDir, sourceDir, file);
};

export const resolveAbsolutePathForFolder = ({
  folder,
  rootDir,
}: {
  folder: string;
  rootDir: string;
}): string => {
  if (path.isAbsolute(folder)) {
    return folder;
  }
  if (path.isAbsolute(rootDir)) {
    return path.join(rootDir, folder);
  }
  return path.resolve(rootDir, folder);
};
