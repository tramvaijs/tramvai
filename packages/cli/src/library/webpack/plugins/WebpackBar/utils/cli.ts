import chalk from 'chalk';

import textTable from 'text-table';
import { range } from '.';

import { BLOCK_CHAR, BLOCK_CHAR2, BAR_LENGTH } from './consts';

export const colorize = (color) => {
  if (color[0] === '#') {
    return chalk.hex(color);
  }

  return chalk[color] || chalk.keyword(color);
};

export const renderBar = (progress, color) => {
  const w = progress * (BAR_LENGTH / 100);
  const bg = chalk.white(BLOCK_CHAR);
  const fg = colorize(color)(BLOCK_CHAR2);

  return range(BAR_LENGTH)
    .map((i) => (i < w ? fg : bg))
    .join('');
};

export function createTable(data) {
  return textTable(data, {
    align: data[0].map(() => 'l'),
  }).replace(/\n/g, '\n\n');
}

export function ellipsis(str, n) {
  if (str.length <= n - 3) {
    return str;
  }
  return `${str.substr(0, n - 1)}...`;
}

export function ellipsisLeft(str, n) {
  if (str.length <= n - 3) {
    return str;
  }
  return `...${str.substr(str.length - n - 1)}`;
}
