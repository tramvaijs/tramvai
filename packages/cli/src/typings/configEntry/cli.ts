import type { Configuration } from 'webpack';
import type { DeduplicateStrategy } from '@tinkoff/webpack-dedupe-plugin';
import type { ConfigEntry, OverridableOption } from './common';

export type TranspilationLoader = 'babel' | 'swc';
export type Minifier = 'terser' | 'swc';

type SimplifiedSvgoConfig = string | { name: string; params: Record<string, any> };
type Notifications = {
  /**
   * @title Defines when success notifications are shown
   */
  suppressSuccess?: false | true | 'always' | 'initial';
  /**
   * @title True to suppress the warning notifications, otherwise false
   */
  suppressWarning?: boolean;
  /**
   * @title True to activate (focus) the terminal window when a compilation error occurs
   */
  activateTerminalOnError?: boolean;
};

export type WebpackExperiments = Omit<
  Configuration['experiments'],
  'buildHttp' | 'lazyCompilation' | 'css'
> & {
  /**
   * @title Enable additional in-memory caching of modules which are unchanged and reference only unchanged modules.
   * @default true
   */
  cacheUnaffected: boolean;
  /**
   * @title Enable backward compatibility with webpack previous major versions
   * @default false
   */
  backCompat: boolean;
};

export type MinicssExperiments = {
  /**
   * @title Use a new webpack API to execute modules instead of child compilers. This improves performance and memory usage a lot.
   * @default true
   */
  useImportModule: OverridableOption<boolean>;
};

export type TranspilationExperiments = {
  /**
   * @title specify loader to transpile js-ts code
   * @default "babel"
   */
  loader: OverridableOption<TranspilationLoader>;
};

export interface Experiments {
  /**
   * @title experiments configuration for [webpack](https://webpack.js.org/configuration/experiments/)
   * @default {}
   */
  webpack: WebpackExperiments;
  /**
   * @title experimental settings for [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin)
   * @default {}
   */
  minicss: MinicssExperiments;

  /**
   * @title experimental settings for code transpilation
   * @default {}
   */
  transpilation: TranspilationExperiments;
  /**
   * @title specify minifer for terser plugin
   * @default "terser"
   */
  minifier: OverridableOption<Minifier>;
}

/**
 * @default {}
 * @additionalProperties true
 */
type Define = Record<string, string>;

/**
 * @tutorial
 * - Common properties that are defined in the tramvai.json config file
 * - These properties are later used to generate the json-schema
 * - Although, some of the marked as required in typings they are anyway converted to optional
 * while generating schema
 * - you can provide default value with "@default" jsdoc in order to replace with default on schema validation
 * - if you provide default value with "@default" mark the property as required to easier its usage in cli code
 * - if the option doesn't have a default value mark it as optional to properly handle this case in cli code
 * - to specify option that might be overridden for dev/prod environments specify its type as `OverridableOption<T>`. You can
 * provide the default value either as single value (will be used in both envs) or in the form of { development: T, production: T } (to be able to
 * provide defaults for all of the envs)
 * - do not use `OverridableOption<T>` with complex types that has internal properties with "@default" as that defaults will not be set by ajv
 */
export interface CliConfigEntry extends ConfigEntry {
  /**
   * @title Enable source maps
   * @default false
   */
  sourceMap: OverridableOption<boolean>;
  /**
   * @title Change different experimental cli settings
   * @default {}
   */
  experiments: Experiments;
  /**
   * @title List of modules to exclude from `@babel/preset-env`
   * @description Option doesn't affect build with swc loader
   */
  excludesPresetEnv?: string[];
  /**
   * @title Webpack's thread-loader config
   * @additionalProperties true
   */
  threadLoader?: Record<string, unknown>;
  /**
   * @title Define static variables that should be replaced by name with provided value
   * @default {}
   */
  define: {
    shared: Define;
    development: Define;
    production: Define;
  };
  /**
   * @title `@deprecated` will be removed in next major release.
   * @default false
   */
  generateDataQaTag: boolean;
  /**
   * @title `@deprecated` will be removed in next major release. Включает использование плагина fill-action-name
   * @default false
   */
  enableFillActionNamePlugin: boolean;
  /**
   * @default {}
   */
  postcss: {
    /**
     * @title Path to postcss config file. By default, `postcss.config.js` file is used
     */
    config?: string;
    /**
     * @title CSS identifiers build algorithm
     */
    cssLocalIdentName?: OverridableOption<string>;
    /**
     * @title Path to postcss config file for assets
     */
    assetsConfig?: string;

    /**
     * @title Enable CSS modules for all files matching /RegExp/i.test(filename) regexp.
     */
    cssModulePattern?: string;
  };
  /**
   * @title `@deprecated` as cli now supports baseUrl and paths from the app's tsconfig.json file.
   * Just check or add configuration to your tsconfig file and remove alias from tramvai.json
   * @additionalProperties true
   */
  alias?: Record<string, any>;
  /**
   * @title svgo-loader options
   */
  svgo?: {
    /**
     * @title svgo plugins, you can find full types in PluginConfig inside `svgo` source code,
     * but this interface too complex and heavy for schema
     */
    plugins?: SimplifiedSvgoConfig[];
  };

  /**
   * @title Settings for image-webpack-loader
   */
  imageOptimization?: {
    /**
     * @title Enable image-webpack-loader
     */
    enabled?: boolean;
    /**
     * @title Pass options to image-webpack-loader
     * @additionalProperties true
     */
    options?: Record<string, any>;
  };
  /**
   * @title transpile libs based only on %40tinkoff/is-modern-lib
   * @default true
   */
  transpileOnlyModernLibs: boolean;
  /**
   * @title Webpack specific settings
   * @default {}
   */
  webpack: {
    /**
     * @title Browser package resolve aliases. E.g. { "stream": "stream-browserify" }
     * @additionalProperties true
     */
    resolveAlias?: Record<string, string>;
    /**
     * @title Browser packages to provide with ProvidePlugin. E.g. { "Buffer": ["buffer", "Buffer"] }
     * @additionalProperties true
     */
    provide?: Record<string, any>;
    /**
     * @title Configure https://webpack.js.org/configuration/watch/#watchoptions
     * @description For OSX users and big projects it's recommended to enable watch polling, e.g. `{ poll: 1000 }`, more info - https://github.com/webpack/watchpack/issues/222
     */
    watchOptions?: Configuration['watchOptions'];
    /**
     * @title Use the specified type of source maps for building in development mode
     */
    devtool: 'eval' | 'eval-cheap-source-map' | 'eval-cheap-module-source-map' | 'eval-source-map';
  };

  // options that affect only production builds
  /**
   * @title Enable DedupePlugin
   * @default {}
   */
  dedupe: {
    /**
     * @title Strategy for DedupePlugin
     * @default true
     */
    enabled: boolean;
    /**
     * @title Does DedupePlugin should be enabled in development mode
     * Why it might be useful see [issue](https://github.com/Tinkoff/tramvai/issues/11)
     * @default false
     */
    enabledDev: boolean;
    /**
     * @title Strategy for DedupePlugin
     * @default "equality"
     */
    strategy: DeduplicateStrategy;
    /**
     * @title Sets ignore to DedupePlugin
     */
    ignore?: string[];
  };
  /**
   * @title terser settings
   * @default {}
   */
  terser: {
    /**
     * @title Configuration for enabling parallel compression for terser plugin
     * @default true
     */
    parallel: boolean;
  };
  /**
   * @title Set minimizer for css
   * @default "css-minimizer"
   */
  cssMinimize: 'csso' | 'css-minimizer';

  // options that affect only development builds

  /**
   * @title React hot-refresh
   * @default {}
   */
  hotRefresh: {
    /**
     * @title Enable react hot-refresh
     * @default true
     */
    enabled?: boolean;
    /**
     * @title Configure react hot-refresh https://github.com/pmmmwh/react-refresh-webpack-plugin#options
     * @default {}
     */
    options?: {
      /**
       * @default false
       */
      overlay?: boolean | Record<string, any>;
      [key: string]: any;
    };
  };
  /**
   * @title Controls build notifications settings during development
   * @default {}
   */
  notifications: Notifications & {
    client?: Notifications;
    server?: Notifications;
  };
  /**
   * @title Specify dependencies that will be shared between application and child-apps
   * @description Properly defining that dependencies may greatly reduce filesize of loaded js on the client
   * @default {}
   */
  shared: {
    /**
     * @title Should default dependencies list be added to shared list
     * @description It includes the list of commonly used dependencies in the child-apps
     * By default, it is enabled in application in case of tramvai/module-child-app is specified in package.json
     * and for child-apps
     */
    defaultTramvaiDependencies?: boolean;
    /**
     * @title add caret range specifier for tramvai dependencies
     * @description minimal versions are inferred from package.json
     * @default true
     */
    flexibleTramvaiVersions: boolean;
    /**
     * @title add chunks to preload as critical in-parallel with "platform.js"
     * @description this option is useful when you need to create async boundary for app dependencies.
     * More info - https://webpack.js.org/concepts/module-federation/#uncaught-error-shared-module-is-not-available-for-eager-consumption
     * @default []
     */
    criticalChunks: string[];
    /**
     * @title list of the dependencies that will be shared
     * @default []
     */
    deps: Array<
      | string
      | {
          /**
           * @title name of the dependency import
           */
          name: string;
          /**
           * @title if dependency is marked as singleton the dependency will be initialized only once and will not be updated
           * @description Do not overuse that feature as it may lead to subtle bugs in case of different versions on different sides
           * @default false
           */
          singleton: boolean;
        }
    >;
  };
}
