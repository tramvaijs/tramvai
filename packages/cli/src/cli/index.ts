import path from 'path';
import { resolvePackageManager, NpmPackageManager } from '@tinkoff/package-manager-wrapper';
import { CLI } from './CLI';
import { Analytics } from '../models/analytics';
import { Logger } from '../models/logger';
import { ConfigManager } from '../models/config';
import { initSentry } from '../utils/sentry';

import type { CommandMap } from '../models/command';

import buildCommand from '../commands/build/command';
import { StartCommand } from '../commands/start/command';
import lintCommand from '../commands/lint/command';
import taskCommand from '../commands/task/command';
import analyze from '../commands/analyze/command';
import generate from '../commands/generate/command';
import newCommand from '../commands/new/command';
import updateCommand from '../commands/update/command';
import addCommand from '../commands/add/command';
import { StaticCommand } from '../commands/static/command';
import { StartProdCommand } from '../commands/start-prod/command';
import { BenchmarkCommand } from '../commands/benchmark/command';

import npmInstallTask from '../tasks/npm-install';
import buildAppTask from '../tasks/build-app';
import type { TaskMap } from '../models/task';
import { getRootFile } from '../utils/getRootFile';
import { getTramvaiConfig } from '../utils/getTramvaiConfig';
import { syncJsonFile } from '../utils/syncJsonFile';

async function loadCommands(): Promise<CommandMap> {
  return [
    buildCommand,
    StartProdCommand,
    StartCommand,
    lintCommand,
    taskCommand,
    analyze,
    generate,
    newCommand,
    updateCommand,
    addCommand,
    StaticCommand,
    BenchmarkCommand,
  ];
}

async function loadTasks(): Promise<TaskMap> {
  return [npmInstallTask, buildAppTask];
}

const defaultPackageInfo = { name: 'init app', version: '0.0.1' };

export async function cliInitialized(cliArgs = process.argv) {
  const logger = new Logger();
  const sentry = initSentry();

  try {
    const commandContext = {
      command: cliArgs[2] || '',
      argv: cliArgs,
    };
    sentry.setContext('command', commandContext);
    const commandsMap = await loadCommands();
    const tasksMap = await loadTasks();

    const { content: packageInfo = defaultPackageInfo } = getRootFile<{
      name: string;
      version: string;
    }>('package.json');
    const { content: config } = getTramvaiConfig();

    const analytic = new Analytics({
      packageInfo,
      trackingCode: 'UA-122261674-1',
    });

    const configManager = new ConfigManager({ config, syncConfigFile: syncJsonFile });
    const packageManager = resolvePackageManager({ rootDir: process.cwd() });

    const cliRootDir = path.resolve(__dirname, '../', '../');
    const cliPackageManager = new NpmPackageManager({
      rootDir: cliRootDir,
    });

    const cliInstance = new CLI(
      commandsMap,
      tasksMap,
      logger,
      configManager,
      analytic,
      cliRootDir,
      cliPackageManager,
      packageManager
    );

    return await cliInstance.run(cliArgs);
  } catch (e: any) {
    logger.event({
      type: 'error',
      event: 'GLOBAL:ERROR',
      message: e.message || e,
      payload: e.details || e.stack,
    });
    sentry.captureException(e);

    throw e;
  } finally {
    await sentry.close();
  }
}
