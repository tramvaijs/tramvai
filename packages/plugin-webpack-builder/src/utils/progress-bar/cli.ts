import c from 'ansis';
import chalk from 'chalk';

import { BLOCK_CHAR, BLOCK_CHAR2, BAR_LENGTH } from './const';
import { range } from './utils';

export const colorize = (color: string) => {
  if (color[0] === '#') {
    return chalk.hex(color);
  }

  return chalk.keyword(color);
};

export const renderBar = (progress: number, color: string) => {
  const w = progress * (BAR_LENGTH / 100);
  const bg = c.white(BLOCK_CHAR);
  const fg = colorize(color)(BLOCK_CHAR2);

  return range(BAR_LENGTH)
    .map((i) => (i < w ? fg : bg))
    .join('');
};

export function ellipsis(str: string, n: number) {
  if (str.length <= n - 3) {
    return str;
  }
  return `${str.slice(0, Math.max(0, n - 1))}...`;
}

export function ellipsisLeft(str: string, n: number) {
  if (str.length <= n - 3) {
    return str;
  }
  return `...${str.slice(str.length - n - 1)}`;
}
