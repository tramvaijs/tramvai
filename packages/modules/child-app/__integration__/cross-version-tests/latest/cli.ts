import { start } from '@tramvai/cli';
import { startCli } from '@tramvai/test-integration';
import { resolve } from 'path';
import detectPort from 'detect-port';

const EXAMPLE_DIR = resolve(__dirname, '..', '..', '..', '..', '..', '..', 'examples', 'child-app');

export const startRootApp = ({
  define,
  env,
}: {
  define: Record<string, string>;
  env: Record<string, string>;
}) => {
  return startCli(
    {
      name: 'root-app',
      type: 'application',
      root: resolve(EXAMPLE_DIR, 'root-app'),
      shared: {
        deps: ['@tramvai/react-query', '@tramvai/module-react-query'],
      },
      define: {
        development: define,
      },
    },
    {
      rootDir: EXAMPLE_DIR,
      env,
      resolveSymlinks: false,
      fileCache: true,
    }
  );
};

export const startChildApp = async (
  name: string,
  {
    shared = {},
  }: {
    shared?: { deps?: string[] };
  } = {}
) => {
  const port = await detectPort(0);
  const staticPort = await detectPort(0);

  return start({
    config: {
      type: 'child-app',
      root: resolve(EXAMPLE_DIR, 'child-apps', name),
      name,
      hotRefresh: {
        enabled: true,
      },
      shared,
    },
    port,
    staticPort,
    rootDir: EXAMPLE_DIR,
    resolveSymlinks: false,
  });
};
