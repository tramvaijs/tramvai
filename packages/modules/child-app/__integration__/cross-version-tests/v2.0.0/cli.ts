import { start } from '@tramvai/cli';
import { startCli } from '@tramvai/test-integration';
import { resolve } from 'path';
import detectPort from 'detect-port';

const EXAMPLE_DIR = __dirname;

export const startRootApp = ({
  define,
  env,
}: {
  define: Record<string, string>;
  env: Record<string, string>;
}) => {
  return startCli(
    {
      name: 'root-app',
      type: 'application',
      root: resolve(EXAMPLE_DIR, 'root-app'),
      commands: {
        build: {
          options: {
            server: resolve(__dirname, 'root-app'),
          },
          configurations: {
            definePlugin: {
              dev: define,
            },
          },
        },
      },
    },
    {
      rootDir: EXAMPLE_DIR,
      env,
      resolveSymlinks: false,
    }
  );
};

export const startChildApp = async (
  name: string,
  {
    shared = {},
  }: {
    shared?: { deps?: string[] };
  } = {}
) => {
  const port = await detectPort(0);
  const staticPort = await detectPort(0);

  return start({
    config: {
      type: 'child-app',
      root: resolve(EXAMPLE_DIR, 'child-apps', name),
      name,
      commands: {
        serve: {
          configurations: {
            hotRefresh: true,
          },
        },
      },
    },
    port,
    staticPort,
    rootDir: EXAMPLE_DIR,
    resolveSymlinks: false,
  });
};
