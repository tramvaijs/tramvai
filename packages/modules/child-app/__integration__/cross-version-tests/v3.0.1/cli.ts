import { start } from '@tramvai/cli';
import { startCli } from '@tramvai/test-integration';
import { resolve } from 'path';
import detectPort from 'detect-port';

const EXAMPLE_DIR = __dirname;

export const startRootApp = ({
  define,
  env,
}: {
  define: Record<string, string>;
  env: Record<string, string>;
}) => {
  return startCli(
    {
      name: 'root-app',
      type: 'application',
      root: resolve(EXAMPLE_DIR, 'root-app'),
      define: {
        development: define,
      },
    },
    {
      rootDir: EXAMPLE_DIR,
      env,
      resolveSymlinks: false,
      fileCache: true,
    }
  );
};

export const startChildApp = async (
  name: string,
  {
    shared = {},
  }: {
    shared?: { deps?: string[] };
  } = {}
) => {
  const port = await detectPort(0);
  const staticPort = await detectPort(0);

  return start({
    config: {
      type: 'child-app',
      root: resolve(EXAMPLE_DIR, 'child-apps', name),
      name,
      hotRefresh: {
        enabled: true,
      },
    },
    port,
    staticPort,
    rootDir: EXAMPLE_DIR,
    resolveSymlinks: false,
  });
};
