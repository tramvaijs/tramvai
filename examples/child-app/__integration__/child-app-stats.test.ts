import path from 'node:path';
import fs from 'node:fs';
import { build } from '@tramvai/cli';

jest.setTimeout(3 * 60 * 1000);

const childAppName = 'header';

describe(`latest stats generation test`, () => {
  beforeAll(async () => {
    await build({
      target: childAppName,
      rootDir: path.resolve(__dirname, '../'),
      fileCache: false,
      env: {
        NODE_ENV: 'production',
      },
    });
  });

  describe('stats', () => {
    it('should build stats.json', async () => {
      const distClientDirectory = path.resolve(__dirname, '../dist', 'child-app');
      const statsJSON = JSON.parse(
        fs.readFileSync(
          path.join(distClientDirectory, `${childAppName}_stats_loadable@0.0.0-stub.json`),
          'utf-8'
        )
      );

      const staticAssets = statsJSON.assets.filter(
        (asset: { name: string }) => !/(js|css)$/.test(asset.name)
      );

      expect(staticAssets.length).toEqual(0);

      delete statsJSON.outputPath;
      delete statsJSON.hash;
      delete statsJSON.publicPath;

      // assets is very unstable for snapshot comparing
      delete statsJSON.assets;

      expect(
        JSON.stringify(statsJSON, null, 2)
          .replace(/_es_js/g, '_js')
          .replace(/_js-_.*"/g, '_js"')
          .replace(/\.chunk\..*\.js"/g, '.chunk.js"')
      ).toMatchSnapshot();
    });
  });
});
