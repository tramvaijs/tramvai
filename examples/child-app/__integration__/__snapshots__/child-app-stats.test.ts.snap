// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`latest stats generation test stats should build stats.json 1`] = `
"{
  "name": "client",
  "assetsByChunkName": {
    "header": [
      "header@0.0.0-stub.css",
      "header_client@0.0.0-stub.js"
    ]
  },
  "namedChunkGroups": {
    "header": {
      "name": "header",
      "chunks": [
        "header"
      ],
      "assets": [
        {
          "name": "header@0.0.0-stub.css"
        },
        {
          "name": "header_client@0.0.0-stub.js"
        }
      ],
      "filteredAssets": 0,
      "assetsSize": null,
      "filteredAuxiliaryAssets": 0,
      "auxiliaryAssetsSize": 0,
      "children": {},
      "childAssets": {}
    }
  },
  "generator": "loadable-components",
  "chunks": [
    {
      "id": "header",
      "files": [
        "header@0.0.0-stub.css",
        "header_client@0.0.0-stub.js"
      ]
    },
    {
      "id": "examples_child-app_child-apps_header_index_ts-examples_child-app_child-apps_header_logo_svg",
      "files": [
        "examples_child-app_child-apps_header_index_ts-examples_child-app_child-apps_header_logo_svg_client.chunk.js"
      ]
    },
    {
      "id": "libs_dippy_lib_di_js",
      "files": [
        "libs_dippy_lib_di_js_client.chunk.js"
      ]
    },
    {
      "id": "tramvai_core_lib_index_js",
      "files": [
        "tramvai_core_lib_index_js"
      ]
    },
    {
      "id": "tramvai_react_lib_react_js",
      "files": [
        "tramvai_react_lib_react_js"
      ]
    },
    {
      "id": "tramvai_core_lib_index_js",
      "files": [
        "tramvai_core_lib_index_js"
      ]
    },
    {
      "id": "tramvai_react_lib_react_js",
      "files": [
        "tramvai_react_lib_react_js"
      ]
    }
  ]
}"
`;
